### PR TITLE
Right click action support.

### PR DESCRIPTION
**Pull request by sli**
Adds support for right click actions on table rows.

## Description
Added an onRowContextMenu prop to the Table component that allows a right click handler to be defined for rows.

## Motivation and Context
Applies a PR from the original fixed-data-table. Addresses #33.

## How Has This Been Tested?
Not formally tested, but a test implementation was created and is stable. In short, this displays the row index in the console as expected when right clicking in a row:

```jsx
<Table
  rowHeight={50}
  rowsCount={rows.length}
  width={1000}
  height={1000}
  headerHeight={50}
  onRowContextMenu={(e, rowIndex) => console.log(`Right clicked row #${rowIndex}.`)}
>
  // ... snip ...
</Table>
```

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Comment by wcjordan** [comment]
Sweet, this looks good to me.  Let me try it out this week, and then we can merge it in.  If it's not too much work could you add an example demonstrating it's use?

File: docs/api-v0.5/TableAPI.md [hunk 1]
```diff
@@ -271,6 +271,13 @@ Callback that is called when a row is double clicked.
 type: `func`
 
 
+### `onRowContextMenu`
```
Comment (KamranAsif) [inline docs/api-v0.5/TableAPI.md]: I don't think you should add this to api-v0.5
This is only being added to 0.6+ right?

**Comment by KamranAsif** [comment]
LGTM

File: docs/api-v0.5/TableAPI.md (hunk 1 above)
Comment (wcjordan) [reply docs/api-v0.5/TableAPI.md]: +1 docs are autogenerated, so we don't need to change this file.

**Comment by difelice** [comment]
I created a new PR addressing your comments: https://github.com/schrodinger/fixed-data-table-2/pull/301

**Comment by wcjordan** [comment]
Closing in favor of #301

**Comment by wcjordan** [comment]
Released with v0.8.11

**Pull request closed without merging**